airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java

```diff
@@ -80,9 +82,12 @@ protected ChangeEvent<String, String> computeNext() {
       // if within the timeout, the consumer could not get a record, it is time to tell the producer to
       // shutdown.
       if (next == null) {
-        LOGGER.info("Closing cause next is returned as null");
-        requestClose();
+        if ((!receivedFirstRecord || hasSnapshotFinished || maxInstanceOfNoRecordsFound >= 10) && !signalledClose) {
```

Comment:
So the thing that happened with carcru's connection. Null was returned during snapshot cause Debezium took more time that 1 minute. We should have kept trying instead of shutting down

Reply:
This should hopefully be addressed by heartbeats